src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,95 @@
-import React from 'react';
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import { Post } from '../types/Post';
+import { Comment } from '../types/Comment';
+import { addComment } from '../services/comments';
+
+type Props = {
+  selectedPost: Post | null;
+  addNewComment: (newComment: Comment) => void;
+  handleSetError: () => void;
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+  selectedPost,
+  addNewComment,
+  handleSetError,
+}) => {
+  const [newComment, setNewComment] = useState({
+    name: '',
+    email: '',
+    comment: '',
+  });
+  const [isLoading, setIsLoading] = useState(false);
+  const [hasInputError, setHasInputError] = useState({
+    name: '',
+    email: '',
+    comment: '',
+  });
+
+  const handleReset = () => {
+    setNewComment({ name: '', email: '', comment: '' });
+    setHasInputError({ name: '', email: '', comment: '' });
+  };
+
+  const handleNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setHasInputError((prevState) => ({ ...prevState, name: '' }));
+    setNewComment((prevComment) => (
+      { ...prevComment, name: event.target.value }));
+  };
+
+  const handleEmailChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setHasInputError((prevState) => ({ ...prevState, email: '' }));
+    setNewComment((prevComment) => (
+      { ...prevComment, email: event.target.value }));
+  };
+
+  const handleCommentChange
+  = (event: React.ChangeEvent<HTMLTextAreaElement>) => {
+    setHasInputError((prevState) => ({ ...prevState, comment: '' }));
+    setNewComment((prevComment) => (
+      { ...prevComment, comment: event.target.value }));
+  };
```

Comment:
```suggestion
  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    const { name, value } = event.target
    setHasInputError((prevState) => ({ ...prevState, [name]: '' }));
    
    setNewComment((prevComment) => (
      { ...prevComment, [name]: value }));
  };
```

instead three handlers you can make one.
something like that